torch/_prims/nvfuser_executor.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+from functools import lru_cache
+
+import torch
+
+from torch.fx import GraphModule
+from torch._prims.utils import getnvFuserDtype, Number
+import torch.overrides
+from torch.utils._pytree import tree_map, tree_flatten, tree_unflatten
+
+if torch.cuda.is_available():
+    from torch._C._nvfuser import DataType, Fusion, FusionDefinition  # type: ignore[import]
+else:
+    DataType = None
+
+
+# nvFuserTensorViewTemplate and nvFuserScalarValTemplate are helper objects
+# for cached construction of the nvFuser's Fusion
+# TODO: change what is stored in the cache for nvFuser's TensorView objects
+# https://github.com/pytorch/pytorch/issues/80551
+@dataclass(frozen=True)
+class nvFuserTensorViewTemplate:
+    size: tuple
+    stride: tuple
+    dtype: DataType
+
+
+@dataclass(frozen=True)
+class nvFuserScalarValTemplate:
```

Comment:
Similarly, I would remove the world `Val` from this name to remove something that references a component of the Fusion IR.